### PR TITLE
[VC] refactor: get whole vc object instead of spec only

### PR DESCRIPTION
**Pull request by zhuangqh**
Signed-off-by: zhuangqh <zhuangqhc@gmail.com>

### CODE REVIEW

File: incubator/virtualcluster/pkg/syncer/util/helper.go [hunk 1]
```diff
@@ -19,34 +19,20 @@ package util
...
 	vc, ok := obj.(*v1alpha1.VirtualCluster)
 	if !ok {
-		return nil, fmt.Errorf("Cannot get the virtualcluster spec from non-vc object.")
-	}
-
-	spec := vc.Spec.DeepCopy()
-	prefixesSet := sets.NewString(spec.OpaqueMetaPrefixes...)
-	if !prefixesSet.Has(constants.DefaultOpaqueMetaPrefix) {
-		spec.OpaqueMetaPrefixes = append(spec.OpaqueMetaPrefixes, constants.DefaultOpaqueMetaPrefix)
+		return nil, fmt.Errorf("cannot get the virtualcluster spec from non-vc object")
```
Comment (Fei-Guo) [inline incubator/virtualcluster/pkg/syncer/util/helper.go:34]: Change the comment accordingly.

**Review by Fei-Guo — COMMENTED** [review]
Can you change the variable name from spec to vc in all relevant cases?

File: incubator/virtualcluster/pkg/syncer/util/helper.go (hunk 1 above)
Comment (zhuangqh) [reply incubator/virtualcluster/pkg/syncer/util/helper.go]: fixed

**Pull request merged**